src/server/conn.rs

```diff
@@ -185,9 +215,11 @@ impl Conn {
             // if registered, we can only remove this flag when
             // writing all data in writable function.
             self.interest.insert(EventSet::writable());
-            return self.reregister(event_loop);
+            try!(self.reregister(event_loop));
         }
 
+        self.on_write_complete = cb;
```

Comment:
Maybe no need now, you can see many other libs like libev libevent also overwrite old callback.
And you can send data, then re-send other in write complete callback again, this is a universal way.